src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx

```diff
@@ -178,14 +180,17 @@ export const TradeInput = memo(({ isCompact }: TradeInputProps) => {
         // but just in case
         return getQuoteErrorTranslation(tradeQuoteError!)
       default:
-        return 'trade.previewTrade'
+        // We got a happy path quote, but we may still be in the context of the demo wallet
+        return !isConnected || isDemoWallet ? 'common.connectWallet' : 'trade.previewTrade'
```

Comment:
prefer we keep predicates in the case for `switch (true)`

```suggestion
      case !isConnected || isDemoWallet:
        // We got a happy path quote, but we may still be in the context of the demo wallet
        return 'common.connectWallet'
      default:
        return 'trade.previewTrade'
```

Reply:
gm 457f5b36c4